frontend/components/AuthGate/AuthGate.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ReactElement } from 'react'
+import { useRouter } from 'next/router'
+import LoadingWrapper from '../LoadingWrapper'
+import { useCurrentUserQuery } from '../../generated/graphql'
+
+/**
+ * Checks that the user is logged in.
+ * If not, redirects them to the login page.
+ */
+
+type AuthGateProps = {
+  children: ReactElement
```

Comment:
nit: it's usually best to default to the `ReactNode` type since it encompasses more possible values that we'd want to use potentially:

```js
type ReactChild = ReactElement | ReactText;
type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```